### ISSUE TITLE
C#: parameter named `extends` is broken in the documentation

### ISSUE THREAD

**Issue by exyi** [doxygen/doxygen#7446]
When I name my parameter `extends`, Doxygen does not write it's name to the output HTML. This is screenshot of the output:
![image](https://user-images.githubusercontent.com/7894687/70907337-857ef400-2000-11ea-930e-8c56434965b4.png)

This is how the declaration look in code:
```C#
public TypeDef(TypeSignature signature, SpecializedType extends, ImmutableArray<SpecializedType> implements, ImmutableArray<MemberDef> members)
```

Doxygen version is `1.8.16` (on Arch Linux, C# 8, .NET Core 3.1.100-preview3-014645)

**Comment by albert-github** [comment]
With just this one line and a default Doxyfile I'm not able reproduce this problem,  also the "Coberec" etc. is not in the shown line, so:
- Can you please attach a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem?  Please don't add external links as they might not be persistent.

**Comment by exyi** [comment]
Ok, I have created trivial repro project: 
[doxygen-repro-extends-parameter.zip](https://github.com/doxygen/doxygen/files/3968316/doxygen-repro-extends-parameter.zip)

Note that the problem is only in this part of the docs:

![image](https://user-images.githubusercontent.com/7894687/70912059-0f33bf00-200b-11ea-925f-971764538b2e.png)

**Comment by albert-github** [comment]
Unfortunately the doxygen configuration file (Doxyfile) is missing, though by mean of a default Doxyfile with just setting `EXTRACT_ALL=YES` I can reproduce the issue.

As constructor I see `public Class1(string extends) { }` and the **question** is now: is `extends` the name of the parameter or is this a language element and has the constructor, in this case no specified parameter?

When having in C++:
```
    class ClassC
    {
        public:
          ClassC(string extends) { };
    }
```
I see a similar behavior.

Looks like that the word `extends` for the type is seen as the `extends` keyword and not as the parameter in `C++` (for `C#` the conclusion is depending on the answer to the question above).

**Comment by exyi** [comment]
Sorry for the missing files, I have fixed that in the post (also the other issue)

> is extends the name of the parameter or is this a language element and has the constructor, in this case no specified parameter?

No, C# does not have a `extends` keyword, this is a legit parameter name.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.17.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).

**Comment by exyi** [comment]
Thank you! I have just checked and it works fine :)
